### DISCUSSION TITLE
Compile time reactivity like Svelte, Solid, Sycamore etc. rather than runtime reactivity(vdom, react etc)

### DISCUSSION THREAD

**Discussion by nikhil-advait** [DioxusLabs/dioxus#339] [General]
First of all a hearty congrats for doing this for the whole rust community. I was first one to follow you guys on discord server and back then I did imagine and hoped that Dioxus will get such a wonderful response from community, all because of your vision. 

General trend in web now a days is towards generating dom patches/reactivity at compile time (svelte, solidjs etc.) rather than at runtime. This has number of benefits, obviously major benefit being performance. Even react is moving towards giving [compile time features](https://www.youtube.com/watch?v=lGEMwh32soc&list=PLNG_1j3cPCaZZ7etkzWA7JfdmKWT0pMsa&index=6). 
In rust ecosystem sycamore/perseus/moonzoon are moving towards compile time reactivity/signals paradigm. Hooks in react served it's purpose but I guess we are finding better ways now to overcome hooks downsides.

I also you saw @jkelleyrtp talking about possibility of having sycamore (or similar style) components along side react style in Dioxus. That would be great. Would you please shed some light upon that vision please.

Again very well done! 
I wish I had more skills and time to contribute but look forward to that in future.

**Comment by raffomania** [comment]
[This blog post](https://dioxuslabs.com/blog/templates-diffing/) explains some of the reasoning behind sticking to the current paradigm. Hope it helps.
